### PR TITLE
fix(dev): kill app server when `remix dev` terminates

### PR DESCRIPTION
**Pull request by pcattori**
Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by pcattori** [comment]
holding off on merging this. need to rebase to `release-next` after v2 prerelease is cut

File: packages/remix-dev/devServer_unstable/index.ts [hunk 1]
```diff
@@ -267,6 +268,12 @@ export let serve = async (
 
   server.listen(options.port);
 
+  exitHook(() => {
+    state.appServer?.pid && kill(state.appServer.pid);
```
Comment (pcattori) [inline packages/remix-dev/devServer_unstable/index.ts:272]: probably want to call `state.appServer.kill()` instead to let the app server terminate gracefully

File: packages/remix-dev/devServer_unstable/index.ts [hunk 2]
```diff
@@ -267,6 +268,12 @@ export let serve = async (
...
+    websocket.close();
+    server.close();
+  });
+
   return new Promise(() => {}).finally(async () => {
```
Comment (ngbrown) [inline packages/remix-dev/devServer_unstable/index.ts:277]: What does this `Promise` do? As far as I've tested it, it never resolves or rejects.

File: packages/remix-dev/devServer_unstable/index.ts (hunk 1 above)
Comment (ngbrown) [inline packages/remix-dev/devServer_unstable/index.ts:272]: Isn't `kill` async?

**Pull request merged**